src/functions.date.js

```diff
@@ -108,28 +108,43 @@ for (let unit in s) {
 	}, {multiValued: true});
 }
 
-_.duration = $.extend(function($this, ms) {
+_.duration = $.extend(function($this, ms, terms = 1) {
 	if (arguments.length === 1) {
 		[ms, $this] = [$this, null];
 	}
-
-	var count = ms || 0;
-	var unit = "ms";
-
-	for (let nextUnit in s) {
-		var nextCount = _.msTo(nextUnit, ms);
-
-		if (nextCount === 0) {
-			break;
+	
+	let unitTime = ms || 0;
+	let timeLeft = ms || 0;
+	
+	if (ms === 0) {
+		terms = 1;
+	}
+	
+	let timeArray = [];
```

Comment:
While in general abbreviations are a bad idea, there is the widespread convention of calling the value that will be eventually returned `ret`, which means that someone reading the code can instantly tell that this is what will eventually be returned, which aids understanding. You may or may not want to follow it. The rest of the codebase mostly does.